File: ruby/difference_of_squares.rb
class Squares
  attr_reader :num_range, :num

  def initialize(num)
    @num=num
    @num_range=(1..num).to_a
  end

  def square_of_sum
    return 0 if num==0
    num_range.reduce(:+) ** 2
  end

  def sum_of_squares
    return 0 if num==0
    num_range.map{|i| i**2}.reduce(:+)
  end

  def difference
    square_of_sum - sum_of_squares
  end
end

File: js/difference-of-squares.js
Squares = function(range){
  this.range = [...Array(range).keys()].map(function(num){return num+1});
  this.sum = this.range.reduce(function(x,y){return x+y})
  this.squares = this.range.map(function(x,y){return x*x})
  this.squareOfSums = Math.pow(this.sum, 2)
  this.sumOfSquares = this.squares.reduce(function(x,y){return x+y})
  this.difference = this.squareOfSums - this.sumOfSquares
};

module.exports = Squares;
